addons/payment_adyen/models/payment_transaction.py

```diff
@@ -375,17 +435,27 @@ def _process_notification_data(self, notification_data):
                     "the void of the transaction with reference %s failed. reason: %s",
                     self.reference, refusal_reason
                 )
-                self._log_message_on_linked_documents(
-                    _("The void of the transaction with reference %s failed.", self.reference)
-                )
-            else:  # 'CAPTURE'
+                if self.source_transaction_id:  # child tx => the event can't be retried
+                    self._set_error(
+                        _("The void of the transaction with reference %s failed.", self.reference)
+                    )
+                else:  # source tx with failed void stays in its state, could be voided again
+                    self._log_message_on_linked_documents(_(
+                        "The void of the transaction with reference %s failed.", self.reference
+                    ))
+            else:  # 'CAPTURE', 'CAPTURE_FAILED'
                 _logger.warning(
                     "the capture of the transaction with reference %s failed. reason: %s",
                     self.reference, refusal_reason
                 )
-                self._log_message_on_linked_documents(
-                    _("The capture of the transaction with reference %s failed.", self.reference)
-                )
+                if self.source_transaction_id:  # child_tx => the event can't be retried
+                    self._set_error(_(
+                        "The capture of the transaction with reference %s failed.", self.reference
+                    ))
+                else:  # source tx with failed capture stays in its state, could be captured again
+                    self._log_message_on_linked_documents(_(
+                        "The capture of the transaction with reference %s failed.", self.reference
+                    ))
```

Comment:
It could be worth merging all these messages into two variables to save some space.